src/routes/v2/authenticated/review.ts

```diff
@@ -0,0 +1,700 @@
+          email: userWithSiteSessionData.email,
+          siteName,
+          requestId,
+        },
+      })
+      return res.status(400).send({
+        message: "Only collaborators of a site can view reviews!",
+      })
+    }
+
+    const possibleReviewRequest = await this.reviewRequestService.getFullReviewRequest(
+      userWithSiteSessionData,
+      site,
+      requestId
+    )
+
+    if (isIsomerError(possibleReviewRequest)) {
+      logger.error({
+        message: "Invalid review request requested",
+        method: "getReviewRequest",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+          requestId,
+        },
+      })
+      return res.status(possibleReviewRequest.status).send({
+        message: possibleReviewRequest.message,
+      })
+    }
+
+    return res.status(200).json({ reviewRequest: possibleReviewRequest })
+  }
+
+  updateReviewRequest: RequestHandler<
+    { siteName: string; requestId: number },
+    ResponseErrorBody,
+    RequestChangeDto,
+    unknown,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    // Step 1: Check that the site exists
+    const { siteName, requestId } = req.params
+    const { userWithSiteSessionData } = res.locals
+    const site = await this.sitesService.getBySiteName(siteName)
+
+    if (!site) {
+      logger.error({
+        message: "Invalid site requested",
+        method: "updateReviewRequest",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+          requestId,
+        },
+      })
+      return res.status(404).send({
+        message: "Please ensure that the site exists!",
+      })
+    }
+
+    // Step 2: Retrieve review request
+    const possibleReviewRequest = await this.reviewRequestService.getReviewRequest(
+      site,
+      requestId
+    )
+
+    if (isIsomerError(possibleReviewRequest)) {
+      logger.error({
+        message: "Invalid review request requested",
+        method: "updateReviewRequest",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+          requestId,
+        },
+      })
+      return res.status(404).json({ message: possibleReviewRequest.message })
+    }
+
+    // Step 3: Check that the user updating is the requestor
+    const { requestor } = possibleReviewRequest
+    if (requestor.email !== userWithSiteSessionData.email) {
+      logger.error({
+        message: "Insufficient to update review request",
+        method: "updateReviewRequest",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+          requestId,
+        },
+      })
+      return res.status(401).json({
```

Comment:
Nit: This should be 403 Forbidden. 401 is when the user is not authenticated and the response would change if the user were to be authenticated.

Reply:
oops, thanks for catching!